apis/risingwave/v1alpha1/risingwave_license.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+// RisingWaveLicense is the license configuration for RisingWave.
+type RisingWaveLicense struct {
+	// SecretName that contains the license. The license must be JWT formatted JSON.
+	// +optional
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
Mark it a required field.

```suggestion
	SecretName string `json:"secretName"`
```

Reply:
Please also clarify the secret key that will be used.

Reply:
> Mark it a required field.

Can you explain the motivation? I learned from

https://github.com/risingwavelabs/risingwave-operator/blob/f7e5e793911d3f25297baca9dcf7e581ff924789/apis/risingwave/v1alpha1/risingwave_ssl.go#L22

Reply:
It probably makes the semantic more clear. The example you quoted seems untuned 😄 

Currently,

```yaml
# valid but has no effect
spec:
  license:
    secretName: ""
```

```yaml
# valid and effective
spec:
  license:
    secretName: "abc"
```

```yaml
# valid and won't be misunderstood
spec:
  # no license field
```

After change, the k8s API service will reject

```yaml
# no longer valid
spec:
  license:
    secretName: ""
```

Reply:
Do you think we need to change the TLS config (`RisingWaveTLSConfiguration`) as well to keep the style consistent?